src/prefect/engine/task_runner.py

```diff
@@ -166,7 +166,11 @@ def initialize_run(  # type: ignore
                 context.update(loop_context)
 
         context.update(
-            task_run_count=run_count, task_name=self.task.name, task_tags=self.task.tags
+            task=self.task,
```

Comment:
Related discussion: https://github.com/PrefectHQ/prefect/issues/889 

We definitely know tasks are serialized by Dask already, but including "heavy" objects in context without a known use case still makes me nervous - context is a `thread.local` object too, so I honestly don't know if that introduces any complications for Tasks which use multithreading or not. It feels like something I'd rather learn during experimentation than from a user though.